kratos/geometries/nurbs_shape_function_utilities/nurbs_curve_shape_functions.h

```diff
@@ -0,0 +1,402 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//
+//  Ported from the ANurbs library (https://github.com/oberbichler/ANurbs)
+//
+
+#if !defined(KRATOS_NURBS_CURVE_SHAPE_FUNCTIONS_H_INCLUDED )
+#define  KRATOS_NURBS_CURVE_SHAPE_FUNCTIONS_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "nurbs_utilities.h"
+
+namespace Kratos {
+///@name Kratos Classes
+///@{
+
+/// NurbsCurveShapeFunction
+/* Container with memory management for an efficient computation
+* of NURBS and B-Spline - 1 dimensional curve functions.
+*/
+class NurbsCurveShapeFunction
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    typedef typename std::size_t IndexType;
+    typedef typename std::size_t SizeType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    NurbsCurveShapeFunction()
+    {
+    };
+
+    /* 
+    * Constructor using the degree u, degree v and the order of shape functions.
+    *   This information is required to make an optimized memory management.
+    */
+    NurbsCurveShapeFunction(
+        const SizeType PolynomialDegree,
+        const SizeType DerivativeOrder)
+    {
+        ResizeDataContainers(PolynomialDegree, DerivativeOrder);
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    double operator()(const IndexType NonzeroControlPointIndex, const IndexType DerivativeRow) const
+    {
+        return ShapeFunctionValue(NonzeroControlPointIndex, DerivativeRow);
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /* 
+    * @brief Resizes the data containers which are needed to compute the
+    *        values of the shape functions.
+    */
+    void ResizeDataContainers(
```

Comment:
should this be in the public interface?

Reply:
Yes, this is used from the shape function generator from the surface shape functions